### PR TITLE
kv: Properly thread context with timeout through kv sendOne

### PR DESCRIPTION
**Pull request by nvb**
The context was available, but we weren't using it before.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5552)

### CODE REVIEW

File: kv/send_test.go [hunk 1]
```diff
@@ -180,8 +180,8 @@ func TestUnretryableError(t *testing.T) {
 		Context:         context.Background(),
 	}
 
-	sendOneFn = func(_ batchClient, _ time.Duration,
-		_ *rpc.Context, done chan batchCall) {
+	sendOneFn = func(_ context.Context, _ *rpc.Context,
+		_ batchClient, done chan batchCall) {
```
Comment (tamird) [inline kv/send_test.go]: un-multiline this and the below now that's shortened

Comment (tamird) [inline kv/send_test.go]: nvm

File: kv/send.go [hunk 2]
```diff
@@ -243,19 +251,12 @@ var sendOneFn = sendOne
...
+func sendOne(ctx context.Context, rpcContext *rpc.Context, client batchClient, done chan batchCall) {
 	addr := client.remoteAddr
 	if log.V(2) {
 		log.Infof("sending request to %s: %+v", addr, client.args)
 	}
 
-	// TODO(tamird/tschottdorf): pass this in from DistSender.
-	ctx := context.TODO()
-	if timeout != 0 {
-		ctx, _ = context.WithTimeout(ctx, timeout)
```
Comment (tamird) [inline kv/send.go]: i think this code should stay here, and that we should `defer cancel()` this context - it was an oversight not to have done that earlier

**Comment by tamird** [comment]
LGTM mod not cancelling the context

**Comment by tschottdorf** [comment]
LGTM mod @tamird's comment

File: kv/send.go (hunk 2 above)
Comment (nvanbenschoten) [inline kv/send.go]: How's that?

**Comment by tamird** [comment]
LGTM

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**